src/process-events.c

```diff
@@ -90,6 +90,19 @@ int kprobe__do_mount(struct pt_regs *ctx)
     return 0;
 }
 
+// returns the pid_tgid() as if it was called by the main thread in
+// the current thread group. This is essentially just the tgid (user
+// space pid) duplicated as the tid == pid in the main thread. This is
+// useful in exec* syscalls since they cause non-main threads to be
+// immediately terminated and the pid will be equal to the tid during
+// the kretprobe even if it wasn't during the kprobe.
+static __always_inline u64 main_thread_pid_tid()
+{
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    u64 pid = pid_tgid >> 32;
+    return pid << 32 | pid;
```

Comment:
we could in theory make this just return pid and have that be the key but since all the other syscalls use the pid_tgid "format" as the keys (and they have to) it would feel weird to have a padded 32 bit number mixed in with the other 64 bit numbers imo.